### PR TITLE
feat(MeetingsAdapter): add MediaPermissions enum

### PR DESCRIPTION
**Pull request by karinasigartau0798**
This enum stored possible values for audio/video permissions.

https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-238895

### CODE REVIEW

File: src/MeetingsAdapter.js [hunk 1]
```diff
@@ -90,6 +90,22 @@ export const MeetingState = {
...
+ * @enum {string}
+ */
+export const MediaPermissions = {
+  UNKNOWN: 'UNKNOWN', // set when the meeting is created
+  ASKING: 'ASKING', // set before asking the user for permission
+  ALLOWED: 'ALLOWED', // set when the promise is fulfilled
+  DENIED: 'DENIED', // set when the promise is rejected because the user has denied access
+  DISMISSED: 'DISMISSED', // promise rejected because the user has dismissed the prompt
+  ERROR: 'ERROR', // promise rejected because of an error
+  IGNORED: 'IGNORED', // if the user clicked the "Proceed without camera" button
```
Comment (cipak) [inline src/MeetingsAdapter.js:106]: if the user has chosen to continue without allowing/denying media access

File: src/MeetingsAdapter.js [hunk 2]
```diff
@@ -90,6 +90,22 @@ export const MeetingState = {
   LEFT: 'LEFT', // the user has left the meeting
 };
 
+/**
+ * Enum for audio/video permissions states.
```
Comment (cipak) [inline src/MeetingsAdapter.js:94]: Enum for media permission states

File: src/MeetingsAdapter.js [hunk 3]
```diff
@@ -90,6 +90,22 @@ export const MeetingState = {
...
+ *
+ * @readonly
+ * @enum {string}
+ */
+export const MediaPermissions = {
+  UNKNOWN: 'UNKNOWN', // set when the meeting is created
```
Comment (cipak) [inline src/MeetingsAdapter.js:100]: let's drop this field and add a note just like for MeetingState: a falsy value means that the permission is unknown (eg: the meeting was just created)

**Review by cipak — APPROVED** [review]

File: src/MeetingsAdapter.js [hunk 4]
```diff
@@ -90,6 +90,22 @@ export const MeetingState = {
...
 
+/**
+ * Enum for media permissions states.
+ * A falsy value for the media permission means that it is unknown (eg: the meeting was just created).
+ *
+ * @readonly
+ * @enum {string}
+ */
+export const MediaPermissions = {
+  ASKING: 'ASKING', // set before asking the user for permission
```
Comment (lalli-flores) [inline src/MeetingsAdapter.js:101]:
```suggestion
  ASKING: 'ASKING', // Set while requesting the user for permission to access media device
```

File: src/MeetingsAdapter.js [hunk 5]
```diff
@@ -90,6 +90,22 @@ export const MeetingState = {
...
+  ALLOWED: 'ALLOWED', // set when the promise is fulfilled
```
Comment (lalli-flores) [inline src/MeetingsAdapter.js:102]:
```suggestion
  ALLOWED: 'ALLOWED', // Set when the user has allowed permission to access media device
```

File: src/MeetingsAdapter.js [hunk 6]
```diff
@@ -90,6 +90,22 @@ export const MeetingState = {
...
+  DENIED: 'DENIED', // set when the promise is rejected because the user has denied access
```
Comment (lalli-flores) [inline src/MeetingsAdapter.js:103]:
```suggestion
  DENIED: 'DENIED', // Set when the user has denied access to media device
```

File: src/MeetingsAdapter.js [hunk 7]
```diff
@@ -90,6 +90,22 @@ export const MeetingState = {
...
+  DISMISSED: 'DISMISSED', // promise rejected because the user has dismissed the prompt
```
Comment (lalli-flores) [inline src/MeetingsAdapter.js:104]:
```suggestion
  DISMISSED: 'DISMISSED', // Set when the user has dismissed the media device access prompt without allowing or denying access
```

File: src/MeetingsAdapter.js [hunk 8]
```diff
@@ -90,6 +90,22 @@ export const MeetingState = {
...
+  ERROR: 'ERROR', // promise rejected because of an error
```
Comment (lalli-flores) [inline src/MeetingsAdapter.js:105]:
```suggestion
  ERROR: 'ERROR', // Set when there is an error while requesting access to media device
```

File: src/MeetingsAdapter.js [hunk 9]
```diff
@@ -90,6 +90,22 @@ export const MeetingState = {
...
+  IGNORED: 'IGNORED', // if the user has chosen to continue without allowing/denying media access
```
Comment (lalli-flores) [inline src/MeetingsAdapter.js:106]:
```suggestion
  IGNORED: 'IGNORED', // Set when the user has chosen to not interact with the device access prompt
```

**Review by lalli-flores — COMMENTED** [review]
Suggestions to make the state explanations a little more generic

**Pull request merged**